packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,89 @@
         default:
           return 'navbar-dark bg-primary'
         }
-      }
+      },
     },
     methods: {
       toggleCollapse (e) {
         e && e.preventDefault()
         this.collapsed = !this.collapsed
+      },
+      // add directory index, e.g http://foo/ or http://foo becomes http://foo/index.html
+      normalizeUrl(url) {
+        if (url.endsWith('.html')) {
+          return url;
+        } else if (url.endsWith('/')) {
+          return `${url}index.html`;
+        } else {
+          return `${url}/index.html`;
+        }
+      },
+      highlightLink(url) {
+        // Performs an exact equality match on links
+        if (this.hasLinkMatch(url, true)) {
+          return;
+        }
+        return this.hasLinkMatch(url, false);
+      },
+      hasLinkMatch(url, isStrict) {
+        const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
+        for (const li of navLis) {
```

Comment:
thanks for implementing this as well! 🙂 

To clarify, the behaviour here is: if any link belonging to the dropdown / navbar item matches, we highlight the dropdown / navbar item as well.

Just a thought, is it possible to:
- apply the `current` styling on `<a>` tags
- additionally apply it to **only** dropdowns?

Reason being we may want to highlight **both** the parent (dropdown header) and child (dropdown item) links

If so maybe we can raise this (and all the other things) in an issue after this pr (unless you want to include it here as well)

Reply:
In view of the upcoming deadline for CP3106, I would like to leave that to another PR 😅